docs/SoftwareArchitecture.md

```diff
@@ -0,0 +1,42 @@
+# Sofware architecture
+
+## Overview
+
+The structural-constraint project develops a new method for identifying compact 3D regions under missense constraint within protein tertiary structures
+
+The project is organised in independent components interacting through defined interfaces
+
+![](softarchitect.png)
+
+## Components
+### Alignment module
+Provides the genomic coordiantes of each nucleotide of each codon for the transect used by a given protein structure.
+#### Interfaces
+- in: gene model, e.g. GTF file for a specific version of GENCODE
+- in: 3D protein structure, e.g. PBR file from AlphaFold 2

```

Comment:
```suggestion
- input: 3D protein structure, e.g. PBR file from AlphaFold 2

```

Reply:
I think this should also be PDB file here